src/hotspot/share/utilities/growableArray.hpp

```diff
@@ -387,6 +387,7 @@ class GrowableArrayWithAllocator : public GrowableArrayView<E> {
 
 public:
   int append(const E& elem) {
+    assert(this->_len != INT_MAX, "Overflow");
```

Comment:
This isn't needed.  `grow` (via `next_power_of_2`) already does the appropriate overflow checking.

Reply:
Then we rely on the underlying growth algorithm always working in power-of-2 steps? What if I want to plug in a different allocator with a different growth cadence?

Reply:
Or, what if we have a simple error? To just accept an overflow here seems dangerous.

That said, I wonder whether we should make the length uintx-sized. As it is now, we are limited to 1G max number of elements.

Reply:
I'm inclined to agree with @kimbarrett . We check length against capacity and grow if needed. The responsibility for checking overflow lies in grow, not append.